### PR TITLE
Upgrade HTAN s3-synapse-sync stack and avoid sandbox roles

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
In this PR, I perform the following: 

- I update the `s3-synapse-sync` stack so I can access the bucket policies
  - This depends on Sage-Bionetworks/s3-synapse-sync#45 as well as a new tag (`v1.6.1`) once it's merged

Depends on Sage-Bionetworks/s3-synapse-sync#45

----

**Edit:** The following changes were reverted and will be made in a separate PR once IT-1967 is settled. 

- I remove Jason Hwee from the bucket policies since he's no longer a Sage employee
- I update Adam's AWS identity to the scicomp `viewer-plus` role (like what I did for mine in #530)
  - This depends on [IT-2049](https://sagebionetworks.jira.com/browse/IT-2049)

### CODE REVIEW

**Review by clarisse-lau — APPROVED** [review]

**Review by adamjtaylor — APPROVED** [review]

File: config/prod/htan-dcc-center-a.yaml
Comment (zaro0508) [inline config/prod/htan-dcc-center-a.yaml:5]: if it helps, i think it's fine to change this to directly reference github instead of s3, https://raw.githubusercontent.com/Sage-Bionetworks/s3-synapse-sync/v1.6.1/s3-synapse-sync-bucket.j2

File: config/prod/htan-dcc-center-a.yaml
Comment (zaro0508) [inline config/prod/htan-dcc-center-a.yaml:22]: adam didn't actually have access to this profile.  I just gave him access in Jumpcloud though.

File: config/prod/htan-dcc-center-a.yaml
Comment (zaro0508) [inline config/prod/htan-dcc-center-a.yaml:5]: if it helps you can just change this to reference the github instead of s3, https://raw.githubusercontent.com/Sage-Bionetworks/s3-synapse-sync/v1.6.1/s3-synapse-sync-bucket.j2

**Review by zaro0508 — APPROVED** [review]

File: config/prod/htan-dcc-center-a.yaml
Comment (zaro0508) [reply config/prod/htan-dcc-center-a.yaml:22]: @BrunoGrandePhD I'm not sure I want to make scicomp an AWS user account store.  I'm wondering whether it would be more appropriate to give @adamjtaylor access to one of your project accounts then use the user ARN of that project account here?

Comment (BrunoGrandePhD) [reply config/prod/htan-dcc-center-a.yaml]: [IT-1967](https://sagebionetworks.jira.com/browse/IT-1967?focusedCommentId=148311) was created to figure this out, but no one from IT has responded in over a month. I made my suggestion there, but it's up to IT to figure out a transition plan away from sandbox roles. 

We don't yet have a production-grade project AWS account for HTAN, and I don't have the bandwidth to create a new AWS account just for housing an IAM role. 

I don't want this to hold up this PR, so I reverted these role changes. This PR is now solely for updating the stack version. We can update the role ARNs once IT-1967 is settled.

**Comment by clarisse-lau** [comment]
Should we update the lambda (s3-synapse-sync.yaml) tag as well?

File: config/prod/htan-dcc-center-a.yaml
Comment (zaro0508) [reply config/prod/htan-dcc-center-a.yaml]: I never said i was completely opposed to using @adamjtaylor scicomp here @BrunoGrandePhD.  It was really just a question and suggestion.   It's your choice to revert but i'm amendable to keeping his scicomp account ARN here if there's not a better option at this time.

Comment (BrunoGrandePhD) [reply config/prod/htan-dcc-center-a.yaml]: I never said that you were completely opposed. I just don't know how long that conversation will take, and I need to debug an issue for a collaborator, who has been waiting since last week. So, I'm uncoupling IT-1967 from this PR, which was primarily motivated to unblock me so I can help our collaborator. 

Please merge if you approve this PR and we'll deal with IT-1967 separately.

**Comment by BrunoGrandePhD** [comment]
@clarisse-lau: Good catch! I've fixed the discrepancy.

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**
